vassal-app/src/main/java/VASSAL/build/module/WizardSupport.java

```diff
@@ -519,20 +519,76 @@ public Component getListCellRendererComponent(JList<?> list, Object value, int i
       return box;
     }
 
+    // From 3.7.6, load Pre-defined setups in the Foreground, not in the Background as arrowing through the PDS drop-down can initiate
+    // multiple concurrent game loads for any but the smallest save files.
     protected void loadSetup(PredefinedSetup setup, final WizardController controller, final Map settings) {
+      controller.setProblem(Resources.getString("WizardSupport.LoadingGame"));
       final GameModule g = GameModule.getGameModule();
+
+      // First check the specified PDS actually exists in the module
+      final String fileName = setup.getFileName();
       try {
-        new SavedGameLoader(controller, settings, new BufferedInputStream(setup.getSavedGameContents()), POST_PLAY_OFFLINE_WIZARD, true).start();
+        if (!g.getDataArchive().contains(fileName)) {
+          controller.setProblem(Resources.getString("WizardSupport.UnableToLoad"));
+          return;
+        }
       }
-      // FIXME: review error message
-      catch (IOException e1) {
+      catch (Exception e) {
         controller.setProblem(Resources.getString("WizardSupport.UnableToLoad"));
+        return;
       }
-      // Predefined Scenarios are files too...
+
+      try {
+        final BufferedInputStream bis = new BufferedInputStream(setup.getSavedGameContents());
```

Comment:
This should be a try-with-resources:
```
try (InputStream in = setup.getSavedGameContents(); 
     BufferedInputStream bis = new BufferedInputStream(in)) {
```
and then you can omit the `bis.close()` later.